internal/wrappers/byor-http.go

```diff
@@ -0,0 +1,78 @@
+package wrappers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	errorConstants "github.com/checkmarx/ast-cli/internal/constants/errors"
+	"github.com/checkmarx/ast-cli/internal/logger"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+const (
+	importsPath       = "/imports"
+	successfulMessage = "The SARIF results were successfully imported into project %s importID: %s"
+)
+
+type ByorHTTPWrapper struct {
+	path          string
+	clientTimeout uint
+}
+
+func NewByorHTTPWrapper(path string) ByorWrapper {
+	return &ByorHTTPWrapper{
+		path:          path,
+		clientTimeout: viper.GetUint(commonParams.ClientTimeoutKey),
+	}
+}
+func (b *ByorHTTPWrapper) Import(projectID, uploadURL string) (string, error) {
+	req := CreateImportsRequest{
+		ProjectID: projectID,
+		UploadURL: uploadURL,
+	}
+
+	jsonBytes, _ := json.Marshal(req)
+	resp, err := SendHTTPRequestWithJSONContentType(http.MethodPost, b.path+importsPath, bytes.NewBuffer(jsonBytes), true, b.clientTimeout)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+	decoder := json.NewDecoder(resp.Body)
+	switch resp.StatusCode {
+	case http.StatusForbidden:
+		return "", getError(decoder, errorConstants.StatusForbidden)
+	case http.StatusUnauthorized:
+		return "", getError(decoder, errorConstants.StatusUnauthorized)
+	case http.StatusInternalServerError:
+		return "", getError(decoder, errorConstants.StatusInternalServerError)
```

Comment:
can it be  “There was a problem importing the SARIF file. Please contact support for further detail with the following error code: “xxxx + API MESSAGE” instead if internal status error?
we can so it in different task if it's complecated

Reply:
It's not complicated, but maybe it's also a good idea to decide on a pattern that will be used in all of out http wrappers when we get error 500?

Reply:
Fixed